src/Lib.P2P.Tests/SwarmTest.cs

```diff
@@ -0,0 +1,1244 @@
+                    "CAASXjBcMA0GCSqGSIb3DQEBAQUAA0sAMEgCQQDlTSgVLprWaXfmxDr92DJE1FP0wOexhulPqXSTsNh5ot6j+UiuMgwb0shSPKzLx9AuTolCGhnwpTBYHVhFoBErAgMBAAE="
+            };
+            var swarmB = new SwarmService {LocalPeer = peerB, NetworkProtector = new OpenNetwork()};
+            await swarmB.StartAsync();
+            var peerBAddress = await swarmB.StartListeningAsync("/ip4/127.0.0.1/tcp/0");
+            Assert.IsTrue(peerB.Addresses.Count() > 0);
+
+            var swarm = new SwarmService {LocalPeer = self, NetworkProtector = new OpenNetwork()};
+            await swarm.StartAsync();
+            try
+            {
+                var remotePeer = await swarm.ConnectAsync(peerBAddress);
+                Assert.AreEqual(2, OpenNetwork.Count);
+            }
+            finally
+            {
+                await swarm.StopAsync();
+                await swarmB.StopAsync();
+            }
+        }
+
+        [TestMethod]
+        public void DeregisterPeer()
+        {
+            var swarm = new SwarmService {LocalPeer = self};
+            swarm.RegisterPeer(other);
+            Assert.IsTrue(swarm.KnownPeers.Contains(other));
+
+            Peer removedPeer = null;
+            swarm.PeerRemoved += (s, e) => removedPeer = e;
+            swarm.DeregisterPeer(other);
+            Assert.IsFalse(swarm.KnownPeers.Contains(other));
+            Assert.AreEqual(other, removedPeer);
+        }
+
+        [TestMethod]
+        public void IsAllowed_Peer()
+        {
+            var swarm = new SwarmService();
+            var peer = new Peer
+            {
+                Id = "QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1h",
+                Addresses = new MultiAddress[]
+                {
+                    "/ip4/127.0.0.1/ipfs/QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1h"
+                }
+            };
+
+            Assert.IsTrue(swarm.IsAllowed(peer));
+
+            swarm.BlackList.Add(peer.Addresses.First());
+            Assert.IsFalse(swarm.IsAllowed(peer));
+
+            swarm.BlackList.Clear();
+            swarm.BlackList.Add("/p2p/QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1h");
+            Assert.IsFalse(swarm.IsAllowed(peer));
+        }
+
+        [TestMethod]
+        public void RegisterPeer_BlackListed()
+        {
+            var swarm = new SwarmService {LocalPeer = self};
+            var peer = new Peer
+            {
+                Id = "QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1h",
+                Addresses = new MultiAddress[]
+                {
+                    "/ip4/127.0.0.1/ipfs/QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1h"
+                }
+            };
+
+            swarm.BlackList.Add(peer.Addresses.First());
+            ExceptionAssert.Throws<Exception>(() => swarm.RegisterPeer(peer));
+        }
+    }
+
+    /// <summary>
+    ///   A noop private network.
+    /// </summary>
+    internal class OpenNetwork : INetworkProtector
+    {
+        public static int Count;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change the visibility of 'Count' or make it 'const' or 'readonly'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)